linkerd/app/inbound/src/http/router/metrics.rs

```diff
@@ -0,0 +1,115 @@
+use linkerd_app_core::{
+    metrics::{
+        prom::{
+            encoding::{self, EncodeLabelSet, LabelSetEncoder},
+            EncodeLabelSetMut,
+        },
+        ServerLabel,
+    },
+    svc::{self, Param},
+    tls::{self, NoServerTls, ServerTls, ServerTlsLabels},
+    transport::{Remote, ServerAddr},
+    Conditional,
+};
+use linkerd_http_prom::{NewCountRequests, RequestCount, RequestCountFamilies};
+
+pub(super) fn layer<N>(
+    request_count: RequestCountFamilies<RequestCountLabels>,
+    // TODO(kate): other metrics families will added here.
+) -> impl svc::Layer<N, Service = NewCountRequests<ExtractRequestCount, N>> {
+    svc::layer::mk(move |inner| {
+        use svc::Layer as _;
+        let extract = ExtractRequestCount(request_count.clone());
+        NewCountRequests::layer_via(extract).layer(inner)
+    })
+}
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub struct RequestCountLabels {
+    protocol: crate::policy::Protocol,
+    server: ServerLabel,
+    server_addr: Remote<ServerAddr>,
+    tls: tls::ConditionalServerTls,
+}
+
+#[derive(Clone, Debug)]
+pub struct ExtractRequestCount(pub RequestCountFamilies<RequestCountLabels>);
+
+// === impl ExtractRequestCount ===
+
+impl<T> svc::ExtractParam<RequestCount, T> for ExtractRequestCount
+where
+    T: Param<Remote<ServerAddr>>
+        + Param<tls::ConditionalServerTls>
+        + Param<crate::policy::AllowPolicy>,
+{
+    fn extract_param(&self, target: &T) -> RequestCount {
+        let Self(family) = self;
+
+        let policy: crate::policy::AllowPolicy = target.param();
+        let server = policy.server_label();
+        let protocol = policy.protocol();
+
+        family.metrics(&RequestCountLabels {
+            protocol,
+            server,
+            server_addr: target.param(),
+            tls: target.param(),
+        })
+    }
+}
+
+// === impl RequestCountLabels ===
+
+impl EncodeLabelSetMut for RequestCountLabels {
+    fn encode_label_set(&self, enc: &mut LabelSetEncoder<'_>) -> std::fmt::Result {
+        use encoding::EncodeLabel as _;
+
+        let Self {
+            server_addr,
+            server: ServerLabel(meta, port),
+            tls,
+            protocol,
+        } = self;
+
+        ("parent_group", meta.group()).encode(enc.encode_label())?;
+        ("parent_kind", meta.kind()).encode(enc.encode_label())?;
+        ("parent_name", meta.name()).encode(enc.encode_label())?;
+        ("parent_port", *port).encode(enc.encode_label())?;
+
+        ("server_addr", server_addr.ip().to_string()).encode(enc.encode_label())?;
+
+        (
+            "protocol",
+            match protocol {
```

Comment:
I don't think this is appropriate here...

What would it mean to have a RequestCount with TLS? Or Detect?

If we're going to document a protocol here, it should reflect the application protocol specifically. It's probably better to omit this.